src/serac/physics/heat_transfer.hpp

```diff
@@ -550,8 +561,15 @@ class HeatTransfer<order, dim, Parameters<parameter_space...>, std::integer_sequ
     auto J_T       = std::unique_ptr<mfem::HypreParMatrix>(jacobian->Transpose());
 
     // If we have a non-homogeneous essential boundary condition, extract it from the given state
-    if (dual_with_essential_boundary) {
-      adjoint_essential = *dual_with_essential_boundary;
+    auto essential_adjoint_temp = duals_with_essential_boundary.find("temperature");
+
+    if (essential_adjoint_temp != duals_with_essential_boundary.end()) {
+      adjoint_essential = essential_adjoint_temp->second;
+    } else {
+      // If the essential adjoint load container does not have a temperature dual but it has a non-zero size, the
+      // user has supplied an incorrectly-named dual vector.
+      SLIC_ERROR_IF(duals_with_essential_boundary.size() != 0,
+                    "Essential adjoint boundary condition given for an unexpected primal field.");
```

Comment:
```suggestion
                    "Essential adjoint boundary condition given for an unexpected primal field. Expected adjoint boundary condition named \"temperature\".");
```

Or something to that extent.